doc/config.md

```diff
@@ -1338,6 +1338,11 @@ messages to the client, but neither will it accept any provided password.
 Set the maximum size of pools for all connections from this user.  If not set,
 the database or `default_pool_size` is used.
 
+### reserve_pool
```

Comment:
Let's stay consistent with the naming of this option and call this `reserve_pool_size` throughout the PR.
```suggestion
### reserve_pool_size
```

Reply:
done!